Products/Zuul/facades/__init__.py

```diff
@@ -206,8 +204,8 @@ def getObjectBrains(self, uid=None, start=0, limit=50, sort='name',
 
 
     def getDevices(self, uid=None, start=0, limit=50, sort='name', dir='ASC',
-                   params=None, hashcheck=None):
-
+                   params={}, hashcheck=None):
+        statuses = params.pop('status', None)
```

Comment:
Let's comment this line as its placement here is important to the performance of this method.  I suggest the following:
```
    # Passing a key that's not an index in the catalog will
    # cause a serious performance penalty.  Remove 'status' from
    # the 'params' parameter before passing it on.
    statuses = params.pop('status', None)
```